modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-test/src/testIntegration/java/com/liferay/headless/commerce/admin/shipment/resource/v1_0/test/ShipmentResourceTest.java

```diff
@@ -259,6 +274,70 @@ private Shipment _addShipment(String externalReferenceCode)
 		return _toShipment(_commerceShipment);
 	}
 
+	private ShippingAddress _randomShippingAddress() throws Exception {
+		CommerceAddress commerceAddress = _commerceOrder.getShippingAddress();
+
+		Country country = commerceAddress.getCountry();
+		Region region = commerceAddress.getRegion();
+
+		return new ShippingAddress() {
+			{
+				city = RandomTestUtil.randomString();
+				countryISOCode = country.getA2();
+				description = RandomTestUtil.randomString();
+				externalReferenceCode = RandomTestUtil.randomString();
+				id = RandomTestUtil.randomLong();
+				latitude = RandomTestUtil.randomDouble();
+				longitude = RandomTestUtil.randomDouble();
+				name = RandomTestUtil.randomString();
+				phoneNumber = RandomTestUtil.randomString();
+				regionISOCode = region.getRegionCode();
+				street1 = RandomTestUtil.randomString();
+				street2 = RandomTestUtil.randomString();
+				street3 = RandomTestUtil.randomString();
+				zip = RandomTestUtil.randomString();
+			}
+		};
+	}
+
+	private void _testPostShipmentWithMoreExternalReferenceCodes()
+		throws Exception {
+
+		Shipment randomShipment = randomShipment();
+
+		ShippingAddress randomShippingAddress = _randomShippingAddress();
+
+		randomShippingAddress.setId(0L);
+
+		long orderId = randomShipment.getOrderId();
+
+		randomShipment.setOrderId(0L);
+		randomShipment.setOrderExternalReferenceCode(
+			_commerceOrder.getExternalReferenceCode());
+		randomShipment.setShippingAddress(randomShippingAddress);
+		randomShipment.setShippingAddressId(0L);
+
+		Shipment postShipment = shipmentResource.postShipment(randomShipment);
+
+		randomShipment.setOrderId(orderId);
+
```

Comment:
I think we can remove this...

Reply:
@smottal I dont think so, then randomshipment will surely have OrderID as zero